main/src/library/Fixpoint/Ram/RamStmt.flix

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Benjamin Dahse
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Fixpoint/Ram {
+    pub enum RamStmt[v] {
+        case Insert(RelOp[v])
+        case Merge(RelSym[v], RelSym[v])
+        case Assign(RelSym[v], RelSym[v])
+        case Purge(RelSym[v])
+        case Seq(List[RamStmt[v]])
+        case Until(List[BoolExp[v]], RamStmt[v])
+        case Comment(String)
+    }
+
+    instance Monoid[RamStmt[v]] {
```

Comment:
No empirical evidence (I haven't profiled compilation specifically). Theoretically it's slow because monoidal fold associates left, which means we end up in the "append case" every time (i.e. folding is quadractic). That can't be good for performance when you have enough facts.